🐛 BUG: wrangler dev should throw the same error message as wrangler publish if no subdomain has been setup yet

### What version of `Wrangler` are you using?

2.0.6

### What operating system are you using?

Mac

### Describe the Bug

In a fresh cloudflare account, run:
```bash
npx wrangler dev
```

what you see:
```js
{
    text: 'Received a bad response from the API',
    notes: [ { text: 'workers.api.error.subdomain_required [code: 10063]' } ],
    location: undefined,
    kind: 'error',
    code: 10063
  }
```

what you should see:
```bash
✘ [ERROR] Error: You need to register a workers.dev subdomain before publishing to workers.dev

  You can either publish your worker to one or more routes by specifying them in wrangler.toml, or
  register a workers.dev subdomain here: <LINK>
```

--- a/.changeset/nine-crabs-raise.md
+++ b/.changeset/nine-crabs-raise.md
@@ -1,0 +1,5 @@
+---
+"wrangler": patch
+---
+
+fix: friendlier error for when a subdomain hasn't been configured in dev mode

--- a/packages/wrangler/src/dev/remote.tsx
+++ b/packages/wrangler/src/dev/remote.tsx
@@ -185,7 +185,21 @@ export function useWorker(props: {
       // we want to log the error, but not end the process
       // since it could recover after the developer fixes whatever's wrong
       if ((err as { code: string }).code !== "ABORT_ERR") {
-        logger.error("Error on remote worker:", err);
+        // instead of logging the raw API error to the user,
+        // give them friendly instructions
+        // for error 10063 (workers.dev subdomain required)
+        if (err.code === 10063) {
+          const errorMessage =
+            "Error: You need to register a workers.dev subdomain before running the dev command in remote mode";
+          const solutionMessage =
+            "You can either enable local mode by pressing l, or register a workers.dev subdomain here:";
+          const onboardingLink = `https://dash.cloudflare.com/${accountId}/workers/onboarding`;
+          logger.error(
+            `${errorMessage}\n${solutionMessage}\n${onboardingLink}`
+          );
+        } else {
+          logger.error("Error on remote worker:", err);
+        }
       }
     });
 

